### PR TITLE
Account for sale prices

### LINKED ISSUE #179: Add sale price support
Account for sale prices on Auto-generated product pages. If product is on sale, then original price will be crossed out in both cases.

On Auto-generated product pages, setting buy buttons with_price has been updated to also take sale price into consideration. If product prices do not have any ranges, then price will not be displayed next to buy button.

### PR DESCRIPTION
**Pull request by piiamatsi**
Account for sale prices on Auto-generated product pages. If product is on sale, then original price will be crossed out in both cases.

On Auto-generated product pages, setting buy buttons with_price has been updated to also take sale price into consideration. If product prices do not have any ranges, then price will not be displayed next to buy button.

Closes #179

### CODE REVIEW

File: components/buy-button.tpl [hunk 1]
```diff
@@ -45,9 +47,7 @@
           <span class="edy-buy-button-title">{{ 'add_to_cart' | lc | escape_once }}</span>
         {% endif -%}
 
-        {%- if with_price and product.price_min_with_tax == product.price_max_with_tax -%}
-          <span class="edy-buy-button-price">{{ product.price_with_tax | money_with_currency: product.currency }}</span>
-        {%- endif -%}
+      <div class="edy-buy-button-price-container"></div>
```
Comment (PeeterPaal) [inline components/buy-button.tpl:50]: Due to buy button structure changes, this div has to be outside of `edy-buy-button` container. Currently prices appear inside buy button on auto-gen product pages.

Screenshot 2023-11-21 at 12 17 18

File: sources/stylesheets/_content-item.scss [hunk 2]
```diff
@@ -255,8 +255,14 @@
...
-    &-divider {
-      margin: 0 4px;
+  .product-price-original {
+    display: inline-block;
+    padding-right: 4px;
+    opacity: .4;
+  }
+
+  .product-price-final {
+    display: inline-block;
```
Comment (PeeterPaal) [inline sources/stylesheets/_content-item.scss:265]: These styles should be indented a bit
```suggestion
    .product-price-original {
      display: inline-block;
      padding-right: 4px;
      opacity: .4;
    }

    .product-price-final {
      display: inline-block;
```

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Prices on auto-gen product page buy buttons are currently displayed inside of the button, which should be fixed. 

Let's also add darkmode styles, as the Front page layout of this design utilizes light- and darkmode styles. Currently product widget prices are hidden when choosing a dark background on Front page.

**Review by tanelj — APPROVED** [review]
Nice and works!

Only remark (old bug): `--button-background-color` variable is missing from Front page layout. Front page includes `template-cs-button-front` but this variable is included only to `template-cs-button`

**Review by PeeterPaal — CHANGES_REQUESTED** [review]
Darkmode styles are missing currently, otherwise looks good!

Front page of this design supports automatically adjusting colors for dark and light backgrounds. Buy button price and product widget prices should use `--light-primary` colour on front page in case of dark background.

Screenshot 2023-11-27 at 10 27 47

**Review by PeeterPaal — APPROVED** [review]
Looks good!

**Pull request merged**